android/app/src/main/java/org/fossasia/openevent/fragments/SpeakersListFragment.java

```diff
@@ -172,9 +174,25 @@ public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case R.id.action_sort:
 
+                int num_orgs=speakersListAdapter.getDistinctOrgs();
```

Comment:
Please refrain from querying database on UI thread. Wrap inside an Observable like other examples from DbSingleton and use that to achieve this :smile: